src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -2248,36 +2249,72 @@ internal bool CheckIfRevisionByDateTimeBeforeAlreadyIncluded(string id, DateTime
         }
 
         public bool CheckIfIdAlreadyIncluded(string[] ids, IEnumerable<string> includes)
+        {
+            foreach (var (_, existsInSession) in GetDocumentsRequiredToLoad(ids, includes))
+            {
+                if (existsInSession == false)
+                    return false;
+            }
+
+            return true;
+        }
+
+        public IEnumerable<(string Id, bool ExistsInSession)> GetDocumentsRequiredToLoad<TEnumerable>(string[] ids, TEnumerable includes)
```

Comment:
This is about document ref includes, what about counters, time series ranges, etc?